### ISSUE TITLE
Scroll position not preserved after clicking "Show more" on a filtered connection list

### ISSUE THREAD

**Issue by courier-new** [sourcegraph/sourcegraph#32302]
- **Sourcegraph version:** Sourcegraph.com
- **Platform information:** Most major browsers (Chrome, Firefox, Safari, Brave tested)

This is easiest to reproduce with the Batch Changes list view but happens anywhere there's a filtered connection with more results than are initially displayed.

#### Steps to reproduce:

1. Visit https://k8s.sgdev.org/batch-changes (signed in to an account in the Sourcegraph org)
2. Scroll the list to the bottom and click "Show more"
3. Wait for query to complete

#### Expected behavior:

URL updates with new `?visible=30` parameter and scroll position on the page does not change.

#### Actual behavior:

URL updates with new `?visible=30` parameter and scroll position is reset to the top of the page.

I believe this is an unintended consequence of https://github.com/sourcegraph/sourcegraph/pull/29178. Since react-scroll-manager caches scroll position against the path, when the filtered connection components change the URL parameter with `history.replace`, the resultant path is not found in the cache. Since there's no scroll position associated with it, then, it defaults to 0, 0 instead, causing the unexpected jump.

The problematic behavior ceases if either:
- `FilteredConnection.useURLQuery` (or `UseConnectionConfig.useURL` for the new hook variation) is set to `false`
- `ScrollManager` is removed from the root rendering tree of `SourcegraphWebApp`

The ideal solution would be to instruct react-scroll-manager to ignore the URL parameter when caching scroll position on routes with filtered connections where changing the parameter does not change what page the user is on.

Here is a screen recording of behavior. Pay attention to the URL when the scroll jump happens (they correspond).

https://user-images.githubusercontent.com/8942601/157168698-509aaf53-d962-40e7-a30a-9eff234cb5d9.mov

**Comment by malomarrec** [comment]
Great issue. Depending on workload, feels like we should prioritize it this sprint.

**Comment by lrhacker** [comment]
From @valerybugakov:

> Based on the quick investigation: as the issue description pointed out, we need to ignore some URL changes (e.g., URL params). The library API doesn't expose props for that. A couple of ideas come to mind:
> 1. Pass wrapped history that would filter out URL changes that we want to ignore to prevent unnecessary scroll restorations. ~0.5d
> 2. Introduce a wrapper around the third-party component with a nicer API, adding a hypothetical `ignoreHistoryUpdates: (location: Location, action: Action) => boolean` prop. ~1d.
>
> Probably relevant lines: [https://github.com/trevorr/react-scroll-manager/blob/master/src/ScrollManager.js#L75-L91](https://www.google.com/url?q=https://github.com/trevorr/react-scroll-manager/blob/master/src/ScrollManager.js%23L75-L91&sa=D&source=docs&ust=1658934106331868&usg=AOvVaw2qLeyEDfa_zuyNqBBVrSpi)

**Comment by mrnugget** [comment]
Dropping in here to underline how important fixing this is and what a brutal impact this issue has on UX. I fixed another bug with `ScrollManager`/`FilteredConnection` yesterday (https://github.com/sourcegraph/sourcegraph/pull/41046) that caused `ScrollManager` to jump back to the top when we refetched a connection, but what I then noticed is _this_ bug here. It's especially frustrating on large lists, such as the repositories page in site-admin area.

Example: this view is really important to our admins https://sourcegraph.com/site-admin/repositories?status=failed-fetch but when they scroll down and click "Show more" they immediately jump back up. And this page requires some scrolling which makes it especially frustrating.
